docs/qos_enforcement.md

```diff
@@ -0,0 +1,254 @@
+  id: string
+  # Determines the IPs that will be allocated to connected subscribers under
+  # this APN service type
+  allocated_ip_range: string
+  # Specify either VLAN or the interface name for backhaul
+  vlan_id: string
+  interface: string
+```
+
+```
+lte_network:
+  id: string
+  name: string
+  description: string
+  ...
+  service_types: array[apn_service_type]
+  ...
+```
+
+```
+feg_lte_network:
+  id: string
+  name: string
+  description: string
+  ...
+  service_types: array[apn_service_type]
+  ...
+```
+
+For both LTE and federated LTE networks, service types must be associated.
+
+```
+subscriber:
+  id: string
+  name: string
+  ...
+  active_base_names: array[base_name]
+  active_policies: array[policy_id]
+  active_apns: array[string]
+  ...
+```
+
+One or more APN objects will be allowed to be associated to subscribers. This is similar to before, though will now create restrictions on associating policies to subscribers.
+
+As before, one or more policies will be allowed to be associated to subscribers. A new limitation will be created where each policy that is associated to a subscriber must share at least one of its QosProfile’s APNs with the subscriber. This must hold true for each policy to subscriber association, and CRUD operations which could break this rule should return an API error and fail to complete.
+
+```
+policy_rule:
+  id: string
+  flow_list: array[flow_description]
+  priority: uint32
+  rating_group: uint32
+  monitoring_key: string
+  ...
+  qos: array[qos_profile]
+```
+
+Multiple QoS objects will be allowed to be associated to each policy object. To implement this, its *qos* field will be changed from the current *flow_qos* type to the new QoS object type, as defined above.
+
+
+### Pipelined Changes for Enforcement
+
+At a high level, changes in `pipelined` are required to support the following:
+* (P1) Enforce QoS via AMBR across the bearer level, APN level, and subscriber level
+* (P2) Load/KPI reporting (to be used for admission control and session lifecycle management by sessiond)
+* (P3) GBR enforcement
+
+
+### Policydb Changes with Streaming
+
+The `policydb` service across the orc8r and gateway requires changes to support QoS enforcement:
+* (P1) Orc8r `policydb` service should be updated to stream to the gateway all the mappings between basenames, policies, QoS profiles, APNs, data plans and subscribers
+* (P1) Gateway `policydb` service should make RAR calls to `sessiond` when updates happen to the policy associations or to policies. To determine when required, policy info and associations are streamed from orc8r, and `policydb `should also query `sessiond` for the relevant info.
+
+
+### Session Manager Changes
+
+The session manager will be changed to support the following:
+* (P1) Provide gRPC methods to retrieve the currently active sessions, including the IMSI, APN, and active policies
+* (P1) Each APN will be assigned a QoS profile to be used for the default bearer
```

Comment:
Will `policyDB` handle the logic of which QoS profile is applied? What needs to be maintained on `SessionD`s side?

Reply:
I think that information should be given from policydb in the `CreateSessionResponse` @ulaskozat